src/nvim/drawscreen.c

```diff
@@ -1092,9 +1092,13 @@ int showmode(void)
   win_T *ruler_win = curwin->w_status_height == 0 ? curwin : lastwin_nofloating();
   if (redrawing() && ruler_win->w_status_height == 0 && global_stl_height() == 0
       && !(p_ch == 0 && !ui_has(kUIMessages))) {
-    grid_line_start(&msg_grid_adj, Rows - 1);
+    if (!ui_has(kUIMessages)) {
+      grid_line_start(&msg_grid_adj, Rows - 1);
```

Comment:
Alternative smaller diff that achieves the same thing at the cost of still calling start() and flush() here:
```diff
@@ -588,6 +588,7 @@ void win_redr_ruler(win_T *wp)
     ADD_C(chunk, INTEGER_OBJ(HLF_MSG));
     assert(attr == HL_ATTR(HLF_MSG));
     ADD_C(content, ARRAY_OBJ(chunk));
+    grid_line_flush();
     ui_call_msg_ruler(content);
     did_show_ext_ruler = true;
   } else {
```